public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilderOptions.tsx

```diff
@@ -7,92 +7,115 @@ import { RadioButtonGroup, Select, AutoSizeInput } from '@grafana/ui';
+import { LokiQuery, LokiQueryType, QueryStats } from '../../types';
 
 export interface Props {
   query: LokiQuery;
   onChange: (update: LokiQuery) => void;
   onRunQuery: () => void;
   maxLines: number;
   app?: CoreApp;
+  datasource: LokiDatasource;
 }
 
-export const LokiQueryBuilderOptions = React.memo<Props>(({ app, query, onChange, onRunQuery, maxLines }) => {
-  const onQueryTypeChange = (value: LokiQueryType) => {
-    onChange({ ...query, queryType: value });
-    onRunQuery();
-  };
-
-  const onResolutionChange = (option: SelectableValue<number>) => {
-    reportInteraction('grafana_loki_resolution_clicked', {
-      app,
-      resolution: option.value,
-    });
-    onChange({ ...query, resolution: option.value });
-    onRunQuery();
-  };
-
-  const onLegendFormatChanged = (evt: React.FormEvent<HTMLInputElement>) => {
-    onChange({ ...query, legendFormat: evt.currentTarget.value });
-    onRunQuery();
-  };
-
-  function onMaxLinesChange(e: React.SyntheticEvent<HTMLInputElement>) {
-    const newMaxLines = preprocessMaxLines(e.currentTarget.value);
-    if (query.maxLines !== newMaxLines) {
-      onChange({ ...query, maxLines: newMaxLines });
+export const LokiQueryBuilderOptions = React.memo<Props>(
+  ({ app, query, onChange, onRunQuery, maxLines, datasource }) => {
+    const [queryStats, setQueryStats] = useState<QueryStats>({ streams: 0, chunks: 0, bytes: 0, entries: 0 });
+    const [prevQuery, setPrevQuery] = useState<LokiQuery>(query);
+
+    const onQueryTypeChange = (value: LokiQueryType) => {
+      onChange({ ...query, queryType: value });
       onRunQuery();
-    }
-  }
+    };
+
+    const onResolutionChange = (option: SelectableValue<number>) => {
+      reportInteraction('grafana_loki_resolution_clicked', {
+        app,
+        resolution: option.value,
+      });
+      onChange({ ...query, resolution: option.value });
+      onRunQuery();
+    };
 
-  let queryType = query.queryType ?? (query.instant ? LokiQueryType.Instant : LokiQueryType.Range);
-  let showMaxLines = isLogsQuery(query.expr);
+    const onLegendFormatChanged = (evt: React.FormEvent<HTMLInputElement>) => {
+      onChange({ ...query, legendFormat: evt.currentTarget.value });
+      onRunQuery();
+    };
+
+    function onMaxLinesChange(e: React.SyntheticEvent<HTMLInputElement>) {
+      const newMaxLines = preprocessMaxLines(e.currentTarget.value);
+      if (query.maxLines !== newMaxLines) {
+        onChange({ ...query, maxLines: newMaxLines });
+        onRunQuery();
+      }
+    }
 
-  return (
-    <EditorRow>
-      <QueryOptionGroup title="Options" collapsedInfo={getCollapsedInfo(query, queryType, showMaxLines, maxLines)}>
-        <EditorField
-          label="Legend"
-          tooltip="Series name override or template. Ex. {{hostname}} will be replaced with label value for hostname."
+    useEffect(() => {
+      if (query.expr === prevQuery.expr) {
+        return;
+      }
```

Comment:
This will always return true, when someone is reloading the page.
I.e. someone entered a query like `{place="luna"}`, reloads the page and they wouldn't see querystats.